// country-state-city/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var country_json_1 = __importDefault(require("./lib/country.json"));
var state_json_1 = __importDefault(require("./lib/state.json"));
var city_json_1 = __importDefault(require("./lib/city.json"));
exports.default = {
    getCountryById: function (id) {
        return _findEntry(country_json_1.default, id);
    },
    getStateById: function (id) {
        return _findEntry(state_json_1.default, id);
    },
    getCityById: function (id) {
        return _findEntry(city_json_1.default, id);
    },
    getStatesOfCountry: function (countryId) {
        var states = state_json_1.default.filter(function (value, index) {
            return value.country_id === countryId;
        });
        return states.sort(compare);
    },
    getCitiesOfState: function (stateId) {
        var cities = city_json_1.default.filter(function (value, index) {
            return value.state_id === stateId;
        });
        return cities.sort(compare);
    },
    getAllCountries: function () {
        return country_json_1.default;
    }
};
var _findEntry = function (source, id) {
    if (id && source != null) {
        var idx = source.findIndex(function (c) { return c.id === id; });
        return (idx !== -1) ? source[idx] : "";
    }
    else
        return "";
};
function compare(a, b) {
    if (a.name < b.name)
        return -1;
    if (a.name > b.name)
        return 1;
    return 0;
}
